dolphin/task_manager/tasks/task.py

```diff
@@ -12,15 +12,48 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import inspect
+
+import decorator
 from oslo_log import log
 
-from dolphin.db.sqlalchemy import api as db
+from dolphin import coordination
+from dolphin import db
+from dolphin import utils
+from dolphin.common import constants
 from dolphin.drivers import api as driverapi
 from dolphin.i18n import _
 
 LOG = log.getLogger(__name__)
 
 
+def sync_task(resource_type):
+
+    @decorator.decorator
+    def _sync_task(f, *a, **k):
```

Comment:
Suggest to set_bit for all resources from the producer side. Producer set all bits at once.

Reply:
Done, please check again